pkg/components/pluggable/discovery_test.go

```diff
@@ -49,6 +64,16 @@ func TestServiceCallback(t *testing.T) {
 	})
 }
 
+func TestConnectionCloser(t *testing.T) {
+	t.Run("connection closer should call grpc close and client reset", func(t *testing.T) {
+		fakeCloser := &fakeGrpcCloser{}
+		fakeService := &fakeReflectService{}
+		closer := reflectServiceConnectionCloser(fakeCloser, fakeService)
+		closer()
+		assert.Equal(t, int64(1), fakeCloser.closeCalled.Load())
+		assert.Equal(t, int64(1), fakeService.resetCalled.Load())
```

Comment:
Quick question should order also be checked as part of the test...because that is what is important.that reset is called before close